### ISSUE TITLE
Fix github action

### ISSUE THREAD

**Issue by dodgyrabbit** [timothycrosley/streamdeck-ui#104]
If anyone has experience to help here that would be appreciated.
Looks like the test requires there to be a display attached but the tests are running headless.
Any ideas how to work around this?

```
ImportError while loading conftest '/home/runner/work/streamdeck-ui/streamdeck-ui/tests/conftest.py'.
tests/conftest.py:3: in <module>
    from streamdeck_ui import api
streamdeck_ui/api.py:11: in <module>
    from pynput.keyboard import Controller, Key
../../../.cache/pypoetry/virtualenvs/streamdeck-ui-HnNvaQR4-py3.8/lib/python3.8/site-packages/pynput/__init__.py:40: in <module>
    from . import keyboard
../../../.cache/pypoetry/virtualenvs/streamdeck-ui-HnNvaQR4-py3.8/lib/python3.8/site-packages/pynput/keyboard/__init__.py:31: in <module>
    backend = backend(__name__)
../../../.cache/pypoetry/virtualenvs/streamdeck-ui-HnNvaQR4-py3.8/lib/python3.8/site-packages/pynput/_util/__init__.py:76: in backend
    raise ImportError('this platform is not supported: {}'.format(
E   ImportError: this platform is not supported: ('failed to acquire X connection: Bad display name ""', DisplayNameError(''))
E
E   Try one of the following resolutions:
E
E    * Please make sure that you have an X server running, and that the DISPLAY environment variable is set correctly
Error: Process completed with exit code 4.
```

**Comment by dodgyrabbit** [comment]
Looks like this may be a workaround:
https://github.com/moses-palmer/pynput/issues/239#issuecomment-609880546

Alternatively, one would need to Mock the Controller (or potentially avoid pynput).

**Comment by tylerjw** [comment]
I found a workaround by using xvfb to have a virtual x11 display.  Lmk if that's an ok solution to this.  The action has some stuff in it for windows/macos but never tests on those platforms.  I'm not sure if this project even works on those platforms?  I don't think the xvfb thing will work on windows or macos but it does fix the github actions run.

Another thing, why are we also running on travis?  Is travis used for more than running the same tests?  Looking at the travis config (which is also currently broken) we can probably just drop travis.

**Comment by dodgyrabbit** [comment]
@tylerjw Would you (or anyone else) have a moment to review the github actions? I'm really not familiar with it and some time ago they all started failing. I think it's because the action is using an old pip version - with known vulnerabilities and failing. Any help would be appreciated.

**Comment by OpenDataAlex** [comment]
@dodgyrabbit  In looking at your latest run, it appears there's an issue with your lint.sh script - specifically with some dependencies/imports.  The github action looks fine overall, as far as I can tell.

**Comment by dodgyrabbit** [comment]
Yea there are some lint items to fix - however - even with those fixed the issue that is left is the lint will fail while doing a security check. Somehow the container/image that is used for the test itself needs to be updated since it's using an old version of pip.

**Comment by OpenDataAlex** [comment]
You can upgrade pip with something akin to:

   - name: Update pip
      run: |
        python -m pip install --upgrade pip

**Comment by dodgyrabbit** [comment]
That's what I tried a while ago
https://github.com/timothycrosley/streamdeck-ui/commit/7aa28ff5fcbc943413af1496cb4d4abe201bb061

**Comment by OpenDataAlex** [comment]
Hmmm. Can you do a run and include a deeper debug on the install lines?
I.e. add -vvv for the pip and poetry lines. I tried looking at the older
run but it got archived out.

**Comment by OpenDataAlex** [comment]
Ooooooooh also, that only upgrades pip2 not pip3. There's a way to do that:

sudo -H pip3 install --upgrade pip

https://stackoverflow.com/questions/38613316/how-to-upgrade-pip3

**Comment by dodgyrabbit** [comment]
The root cause here was a little tricky so writing here for posterity sake.

I knew the problem was during the lint.sh stage the `safety` step detects an old version of pip.

Here is what I did to find root cause:
1. Install the latest pip in a step 2e410df88f03d484d3e25f3e453a55b8c0fbb0a1
Output said "dependency was already met". Scratch head.

2. Output the version in a step 859463d2d995ac555c7646f2de2b1025c1b08c3c
Output confirmed we're already on latest
This was puzzling, why would `safety` fail? Conclusion: They must "see" different versions, but why?

3. This [thread](https://github.com/python-poetry/poetry/issues/732#issuecomment-447082963) made me realize the poetry installs pip into its own venv and therefore ignores the global version installed in Ubuntu. Solution, check poetry install version.

4. Don't hard code the version of poetry in cd0b0614ae4566591b25626bbd49119b1c629481
Latest version has a safe version of pip. Small downside is if a new version of poetry breaks things, it could in future break the pipeline but probably better to stay current.

5. Fix the remaining tests in e8cc045af164da69635e936fe61153f0fe5949d7
